crates/paths/src/lib.rs

```diff
@@ -0,0 +1,87 @@
+//! The spacetimedb directory tructure, represented as a type hierarchy.
+
+use crate::utils::PathBufExt;
+
+pub mod cli;
+pub mod server;
+pub mod standalone;
+mod utils;
+
+#[doc(hidden)]
+pub use serde as __serde;
+
+path_type! {
+    /// The --root-dir for the spacetime installation, if specified.
+    RootDir: dir
+}
+
+impl RootDir {
+    pub fn cli_config_dir(&self) -> cli::ConfigDir {
+        cli::ConfigDir(self.0.join("config"))
+    }
+
+    pub fn cli_bin_file(&self) -> cli::BinFile {
+        let mut path = self.0.join("spacetime");
+        path.set_extension(std::env::consts::EXE_EXTENSION);
+        cli::BinFile(path)
+    }
+
+    pub fn cli_bin_dir(&self) -> cli::BinDir {
+        cli::BinDir(self.0.join("bin"))
+    }
+
+    pub fn data_dir(&self) -> server::ServerDataDir {
+        server::ServerDataDir(self.0.join("data"))
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct SpacetimePaths {
+    pub cli_config_dir: cli::ConfigDir,
+    pub cli_bin_file: cli::BinFile,
+    pub cli_bin_dir: cli::BinDir,
+    pub data_dir: server::ServerDataDir,
+}
+
+impl SpacetimePaths {
+    /// Get the default directories for the currrent platform.
+    ///
+    /// Returns an error if the platform director(y/ies) cannot be found.
+    pub fn platform_defaults() -> anyhow::Result<Self> {
+        #[cfg(windows)]
+        {
+            let data_dir = dirs::data_local_dir().ok_or_else(|| anyhow::anyhow!("Could not find LocalAppData"))?;
+            let root_dir = RootDir(data_dir.joined("SpacetimeDB"));
+            Ok(Self::from_root_dir(&root_dir))
+        }
```

Comment:
the `dirs` crate says that it works with the XDG directories under Linux - is there a reason we have to manually switch to XDG directories when not on windows?

Reply:
also does this work properly on MacOS?

Reply:
I'm not able to test on macOS, but it should. The reason we use this (~~and there's a comment saying this right above here~~ oops - there used to be) is because `dirs` doesn't use the xdg dirs on macOS, which the proposal says we should.